loadable_apps/micomapp/micomapp.c

```diff
@@ -22,16 +22,86 @@
 
 #include <stdio.h>
 #include <unistd.h>
+#ifdef CONFIG_EXAMPLES_MICOM_TIMER_TEST
+#include <string.h>
+#endif
 #ifdef CONFIG_BINARY_MANAGER
 #include <binary_manager/binary_manager.h>
 #endif
 #include "micomapp_internal.h"
 
+#ifdef CONFIG_EXAMPLES_MICOM_TIMER_TEST
+#define TIMER_ARG_MAX 10
+#define TIMER_ARG_NUM 5
+
+char *timer_args[TIMER_ARG_NUM];
+static void alloc_timer_args(void)
+{
+	timer_args[0] = (char *)malloc(strlen("timer") + 1);
+	if (timer_args[0] == NULL) {
+		goto timer_errout0;
+	}
+	strncpy(timer_args[0], "timer", strlen("timer") + 1);
+
+	timer_args[1] = (char *)malloc(strlen("-n") + 1);
+	if (timer_args[1] == NULL) {
+		goto timer_errout1;
+	}
+	strncpy(timer_args[1], "-n", strlen("-n") + 1);
+
+	timer_args[2] = (char *)malloc(strlen(CONFIG_EXAMPLES_MICOM_TIMER_REPEAT) + 1);
+	if (timer_args[2] == NULL) {
+		goto timer_errout2;
+	}
+	strncpy(timer_args[2], CONFIG_EXAMPLES_MICOM_TIMER_REPEAT, strlen(CONFIG_EXAMPLES_MICOM_TIMER_REPEAT) + 1);
+
+	timer_args[3] = (char *)malloc(strlen("-t") + 1);
+	if (timer_args[3] == NULL) {
+		goto timer_errout3;
+	}
+	strncpy(timer_args[3], "-t", strlen("-t") + 1);
+
+	timer_args[4] = (char *)malloc(strlen(CONFIG_EXAMPLES_MICOM_TIMER_INTERVAL) + 1);
+	if (timer_args[4] == NULL) {
+		goto timer_errout4;
+	}
+	strncpy(timer_args[4], CONFIG_EXAMPLES_MICOM_TIMER_INTERVAL, strlen(CONFIG_EXAMPLES_MICOM_TIMER_INTERVAL) + 1);
+
+	return OK;
+
+timer_errout4:
+	free(timer_args[3]);
+timer_errout3:
+	free(timer_args[2]);
+timer_errout2:
+	free(timer_args[1]);
+timer_errout1:
+	free(timer_args[0]);
+timer_errout0:
+	printf("Out of Memory for testing Timer.\n");
+	return ERROR;
+}
+
+static void free_timer_args(void)
+{
+	int arg_idx;
+	for (arg_idx = 0; arg_idx < TIMER_ARG_NUM; arg_idx++) {
+		free(timer_args[arg_idx]);
```

Comment:
it is always good to have null check before free
if(timer_args[arg_idx] != NULL)
       free(timer_args[arg_idx]);